app/client/src/contexts/Glossary.tsx

```diff
@@ -1,29 +1,61 @@
-// @flow
-
-import React, { createContext, useEffect, useState } from 'react';
-import type { Node } from 'react';
+import { createContext, useEffect, useState } from 'react';
+import type { ReactNode } from 'react';
 // contexts
 import { useServicesContext } from 'contexts/LookupFiles';
 // utilities
 import { proxyFetch } from 'utils/fetchUtils';
 
+// types
+interface GlossaryData {
+  term: string;
+  definition: string;
+}
+
+type GlossaryStatus = 'fetching' | 'failure' | 'success';
+
+interface FetchSuccessState {
+  status: 'success';
+  data: GlossaryData[];
+}
+
+interface FetchFailureState {
+  status: 'failure';
+  data: never[];
+}
+
+type FetchState = FetchSuccessState | FetchFailureState;
+
+declare global {
+  interface Window {
+    fetchGlossaryTerms: Promise<FetchState>;
+  }
+}
+
 // --- components ---
-const GlossaryContext: Object = createContext({
+type GlossaryContextProps = {
+  initialized: boolean;
+  setInitialized: (initialized: boolean) => void;
+  glossaryStatus: GlossaryStatus;
+  setGlossaryStatus: (status: GlossaryStatus) => void;
```

Comment:
```suggestion
  initialized: boolean;
  setInitialized: Dispatch<SetStateAction<boolean>>;
  glossaryStatus: GlossaryStatus;
  setGlossaryStatus: Dispatch<SetStateAction<GlossaryStatus>>;
```

This should be using the React types.